src/transport/SecureSession.h

```diff
@@ -162,7 +162,8 @@ class SecureSession : public Session, public ReferenceCounted<SecureSession, Sec
         switch (mPeerAddress.GetTransportType())
         {
         case Transport::Type::kUdp:
-            return GetRemoteMRPConfig().mIdleRetransTimeout * (CHIP_CONFIG_RMP_DEFAULT_MAX_RETRANS + 1);
+            return GetRetransmissionTimeout(mRemoteMRPConfig.mActiveRetransTimeout, mRemoteMRPConfig.mIdleRetransTimeout,
+                                            GetLastPeerActivityTime(), kMinActiveTime);
```

Comment:
This addresses the first problem I identified in #23328, but without the fix to the second, this change on this line will now actually cause back-to-back transactions to sleepy devices to more likely fail.

While previous to this change, we would be selecting the wrong interval for retransmission, the timeout on the exchange will be computed pseudo correctly based on the IDLE interval.

With this change, the exchange timeout will be set quite low.

On the whole, I'm fine if we merged this, as long as we fast-follow with a fix to the second problem as well.

Reply:
I will investigate in a followup how to fix the `IsPeerActive` condition. Initially this patch was changing the `IsPeerActive` logic but I have reverted those changes to not makes even more less spec-compliant.